packages/components/src/form-file-upload/types.ts

```diff
@@ -27,8 +27,11 @@ export type FormFileUploadProps = {
 	icon?: ComponentProps< typeof Icon >[ 'icon' ];
 	/**
 	 * Whether to allow multiple selection of files or not.
+	 *
+	 * This property will also accept a string with the value `add` to allow multiple
+	 * selection of files without the need to use the `Shift` or `Ctrl`/`Cmd` keys.
 	 */
-	multiple?: InputHTMLAttributes< HTMLInputElement >[ 'multiple' ];
+	multiple?: InputHTMLAttributes< HTMLInputElement >[ 'multiple' ] | string;
```

Comment:
I don't think it makes sense to add this change to `FormFileUpload` because an `'add'` value will not change the component behavior at all.

I believe we want to be casting the `'add'` value to a boolean at the point where it's passed to the FormFileUpload, not inside FormFileUpload. For example [here](https://github.com/WordPress/gutenberg/blob/c14792d535c372a045ff4fc262fcaca43c0326a6/packages/block-editor/src/components/media-placeholder/index.js#L472).

Reply:
This makes sense, thank you. I wasn't sure about considering a string value in the `FormFileUpload` component.

Could you explain a little further about casting the `'add'` value to a boolean? I was thinking something like this:

```
if ( multiple === 'add' ) multiple = true;
```

But I'm wondering if we need to handle it at all, as I think a non-empty string will be read as `true` by default.

Reply:
> Could you explain a little further about casting the `'add'` value to a boolean?

Sure!

https://github.com/WordPress/gutenberg/blob/c14792d535c372a045ff4fc262fcaca43c0326a6/packages/block-editor/src/components/media-placeholder/index.js#L472

We could do for example `multiple={ !! multiple }` on this line. We don't "need" to handle it in the sense that it would work fine in runtime, but if at any point we start type checking this file it would be a type error. So I would recommend explicitly casting to boolean.

Reply:
Ah yep OK, that works for me!